### PR TITLE
Make builds deterministic with ContinuousIntegrationBuild flag

### PR DESCRIPTION
**Pull request by lahma**
Currently [NuGet Package Explorer gives a warning](https://nuget.info/packages/NUnit/4.0.1) about missing compilation flags. This can be fixed by setting [ContinuousIntegrationBuild](https://cakebuild.net/api/Cake.Common.Tools.DotNet.MSBuild/DotNetMSBuildSettings/29A05B4C) to true. Also updated workflow definition to use new version of tasks to get rid of warnings during CI runs.

image

### CODE REVIEW

**Comment by lahma** [comment]
After the changes now complaints for download CI artifacts
image

**Comment by OsirisTerje** [comment]
This flag should be set when build on the server, but not locally.  So it needs to be dropped in as a parameter, with default false.

File: build.cake [hunk 1]
```diff
@@ -116,7 +116,7 @@ DotNetBuildSettings CreateDotNetBuildSettings() =>
         Configuration = configuration,
         NoRestore = true,
         Verbosity = DotNetVerbosity.Minimal,
-        MSBuildSettings = new DotNetMSBuildSettings { Version = packageVersion }
+        MSBuildSettings = new DotNetMSBuildSettings { Version = packageVersion, ContinuousIntegrationBuild = true }
```
Comment (OsirisTerje) [inline build.cake:119]: This flag should be set when build on the server, but not locally. So it needs to be dropped in as a parameter, with default false.

**Review by OsirisTerje — CHANGES_REQUESTED** [review]

File: build.cake (hunk 1 above)
Comment (lahma) [reply build.cake]: I changed the flag to be based on [BuildSystem.GitHubActions.IsRunningOnGitHubActions](https://cakebuild.net/api/Cake.Common.Build.GitHubActions/IGitHubActionsProvider/3F54CBE7)

**Review by OsirisTerje — APPROVED** [review]

**Comment by OsirisTerje** [comment]
Thanks @lahma !

**Pull request merged**